cloudwatchlogs-with-dlq/samplingutils.js

```diff
@@ -0,0 +1,38 @@
+function sampleTraffic(samplingPercent, records) {
+    if (!samplingPercent) {
+        return records;
+    }
+    var filteredRecords = [];
+    var toBeSamplesRecords = [];
+    records.forEach(function (log) {
+        if ("logLevel" in log.message.message) {
+            if(Number.isInteger(log.message.message.logLevel)){
+                if (log.message.message.logLevel >= 4) {
+                    filteredRecords.push(log);
+                } else {
+                    toBeSamplesRecords.push(log);
+                }
+            } else {
+                filteredRecords.push(log);
+            }
+        } else {
+            filteredRecords.push(log);
+        }
+    });
+    if(samplingPercent == 0) {
+        var increment = 0;
+    } else {
+        var increment = 100/samplingPercent;
```

Comment:
let's type cast this to integer so that increment is a natural number always.
Also I think we should have a condition samplingPercent > 0 and <= 100 and then we can run a for loop inside it.
Also would be good to break this into two separate functions one for filtering and other for sampling. Filtering one returns filtered + tobesampled records and takes records as input
while sampling one takes tobesampled records and sampling percent as input and returns sampled records
later on we merge the filtered and sampled records